File: Python/Clothing.py
from Product import Product


class Clothing(Product):
    def __init__(self, idProduct, name, brand, price, size, material, gender):
        super().__init__(idProduct, name, brand, price)
        self.size = size
        self.material = material
        self.gender = gender

    def get_size(self):
        return self.size

    def set_size(self, size):
        self.size = size

    def get_material(self):
        return self.material

    def set_material(self, material):
        self.material = material

    def get_gender(self):
        return self.gender

    def set_gender(self, gender):
        self.gender = gender

    def __str__(self):
        return super().__str__() + " {:<5s}| {:<10s}| {:<6s} |".format(self.size, self.material, self.gender)


# Contoh penggunaan kelas Clothing
# clothing = Clothing(1, "Clothing 1", "Brand 1", 20.75, "M", "Cotton", "Male")
# print(clothing)

File: CPP/Clothing.cpp
#include <iostream>
#include <string>
#include "Product.cpp"

using namespace std;

// class Product
// {
// protected:
//     int idProduct;
//     string name;
//     string brand;
//     double price;

// public:
//     Product(int idProduct, string name, string brand, double price)
//     {
//         this->idProduct = idProduct;
//         this->name = name;
//         this->brand = brand;
//         this->price = price;
//     }

//     string toString()
//     {
//         return to_string(idProduct) + "| " + name + "| " + brand + "| " + to_string(price);
//     }
// };

class Clothing : public Product
{
private:
    string size;
    string material;
    string gender;

public:
    Clothing(int idProduct, string name, string brand, double price, string size, string material, string gender)
        : Product(idProduct, name, brand, price)
    {
        this->size = size;
        this->material = material;
        this->gender = gender;
    }

    string getSize()
    {
        return size;
    }

    void setSize(string size)
    {
        this->size = size;
    }

    string getMaterial()
    {
        return material;
    }

    void setMaterial(string material)
    {
        this->material = material;
    }

    string getGender()
    {
        return gender;
    }

    void setGender(string gender)
    {
        this->gender = gender;
    }

    string toString()
    {
        return Product::toString() + "| " + size + "| " + material + "| " + gender;
    }
};

// int main()
// {
//     Clothing clothing(1, "Shirt", "Brand", 50000, "L", "Cotton", "Male");
//     cout << clothing.toString() << endl;

//     return 0;
// }